reactor-core/src/jcstress/java/reactor/core/publisher/FluxReplayStressTest.java

```diff
@@ -0,0 +1,684 @@
+			r.r4 = subscriber2.onCompleteCalls.get();
+			r.r5 = subscriber1.onErrorCalls.get();
+			r.r6 = subscriber2.onErrorCalls.get();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {
+			"10, 10, 1, 1, 0, 0"}, expect = ACCEPTABLE, desc = "concurrent subscription succeeded")
+	@State
+	public static class RefCntGraceConcurrentSubscriptionRangeNoneFusionStressTest
+			extends RefCntConcurrentSubscriptionBaseStressTest<Integer> {
+
+		public RefCntGraceConcurrentSubscriptionRangeNoneFusionStressTest() {
+			super(Flux.range(0, 10)
+			          .hide(), Duration.ofSeconds(1));
+		}
+
+		@Actor
+		public void subscribe1() {
+			sharedSource.subscribe(subscriber1);
+		}
+
+		@Actor
+		public void subscribe2() {
+			sharedSource.subscribe(subscriber2);
+		}
+
+		@Arbiter
+		public void arbiter(IIIIII_Result r) {
+			r.r1 = subscriber1.onNextCalls.get();
+			r.r2 = subscriber2.onNextCalls.get();
+			r.r3 = subscriber1.onCompleteCalls.get();
+			r.r4 = subscriber2.onCompleteCalls.get();
+			r.r5 = subscriber1.onErrorCalls.get();
+			r.r6 = subscriber2.onErrorCalls.get();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {
+			"0, 0, 1, 1, 0, 0"}, expect = ACCEPTABLE, desc = "concurrent subscription succeeded")
+	@State
+	public static class RefCntGraceConcurrentSubscriptionEmptyAsyncStressTest
+			extends RefCntConcurrentSubscriptionBaseStressTest<Object> {
+
+		public RefCntGraceConcurrentSubscriptionEmptyAsyncStressTest() {
+			super(Flux.empty()
+			          .publishOn(Schedulers.immediate()), Duration.ofSeconds(1));
+		}
+
+		@Actor
+		public void subscribe1() {
+			sharedSource.subscribe(subscriber1);
+		}
+
+		@Actor
+		public void subscribe2() {
+			sharedSource.subscribe(subscriber2);
+		}
+
+		@Arbiter
+		public void arbiter(IIIIII_Result r) {
+			r.r1 = subscriber1.onNextCalls.get();
+			r.r2 = subscriber2.onNextCalls.get();
+			r.r3 = subscriber1.onCompleteCalls.get();
+			r.r4 = subscriber2.onCompleteCalls.get();
+			r.r5 = subscriber1.onErrorCalls.get();
+			r.r6 = subscriber2.onErrorCalls.get();
+		}
+	}
+
+	@JCStressTest
+	@Outcome(id = {
+			"0, 0, 1, 1, 0, 0"}, expect = ACCEPTABLE, desc = "concurrent subscription succeeded")
+	@State
+	public static class RefCntGraceConcurrentSubscriptionEmptyNoneStressTest
+			extends RefCntConcurrentSubscriptionBaseStressTest<Object> {
+
+		public RefCntGraceConcurrentSubscriptionEmptyNoneStressTest() {
+			super(Flux.empty()
+			          .hide(), Duration.ofSeconds(1));
+		}
+
+		@Actor
+		public void subscribe1() {
+			sharedSource.subscribe(subscriber1);
+		}
+
+		@Actor
+		public void subscribe2() {
+			sharedSource.subscribe(subscriber2);
+		}
+
+		@Arbiter
+		public void arbiter(IIIIII_Result r) {
+			r.r1 = subscriber1.onNextCalls.get();
+			r.r2 = subscriber2.onNextCalls.get();
+			r.r3 = subscriber1.onCompleteCalls.get();
+			r.r4 = subscriber2.onCompleteCalls.get();
+			r.r5 = subscriber1.onErrorCalls.get();
+			r.r6 = subscriber2.onErrorCalls.get();
+		}
+	}
```

Comment:
Is there any reason why an error scenario is not tested for `FluxRefCountGrace`?

Reply:
I added the missing scenarios.